sympy/solvers/ode/single.py

```diff
@@ -232,14 +238,17 @@ def _matches(self):
         if order != 1:
             return False
 
-        eq = expand(eq / eq.coeff(df))
-        eq = eq.collect(f(x), func = cancel)
-
         pattern = self._equation(f(x), x, 1)
 
+        if self.simplify:
+            eq = expand(eq / eq.coeff(df))
+        eq = eq.collect(f(x), func = cancel)
```

Comment:
Maybe instead of having a simplify flag this could check what the coefficient of df is in pattern before dividing. If the pattern has a coefficient with no Wilds then we need to divide/multiply the equation to match that.

Reply:
I think maybe it should be `eq = expand(eq * (pattern.coeff(df) / eq.coeff(df)))`.